src/Glimpse.Server.Web/Resources/RequestHistory.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Glimpse.Server.Web.Extensions;
+using Microsoft.AspNet.Http;
+using Microsoft.Net.Http.Headers;
+
+namespace Glimpse.Server.Web.Resources
+{
+    public class RequestHistory : IResource
+    {
+        private readonly IQueryRequests _requests;
+
+        public RequestHistory(IStorage storage)
+        {
+            if (!storage.Supports<IQueryRequests>())
+                throw new ArgumentException($"IStorage implementation of type '{storage.GetType().FullName}' does not support IQueryRequests.", nameof(storage));
+
+            _requests = storage.As<IQueryRequests>();
+        }
+
+        // TODO: The parameter parsing in this method is brittle and will require more work
+        public async Task Invoke(HttpContext context, IDictionary<string, string> parameters)
+        {
+            var response = context.Response;
+            response.Headers[HeaderNames.ContentType] = "application/json";
+            
+            var filters = new RequestFilters();
+            if (parameters.ContainsKey("dmin"))
+                filters.DurationMinimum = float.Parse(parameters["dmin"]);
+
+            if (parameters.ContainsKey("dmax"))
+                filters.DurationMaximum = float.Parse(parameters["dmax"]);
+
+            if (parameters.ContainsKey("url"))
+                filters.UrlContains = parameters["url"];
+
+            if (parameters.ContainsKey("methods"))
+                filters.MethodList = parameters["methods"].Split(',');
+
+            if (parameters.ContainsKey("smin"))
+                filters.StatusCodeMinimum = int.Parse(parameters["smin"]);
+
+            if (parameters.ContainsKey("smax"))
+                filters.StatusCodeMaximum = int.Parse(parameters["smax"]);
+
+            if (parameters.ContainsKey("tags"))
+                filters.TagList = parameters["tags"].Split(',');
+
+            if (parameters.ContainsKey("user"))
+                filters.UserId = parameters["user"];
+
+            if (parameters.ContainsKey("before"))
+                filters.RequesTimeBefore = DateTime.Parse(parameters["before"]);
+
+            IEnumerable<string> list;
+            if (parameters.ContainsKey("types"))
+            {
+                var types = parameters["types"].Split(',');
+                list = await _requests.Query(filters, types);
+            }
+            else
+            {
+                list = await _requests.Query(filters);
+            }
+
+            await response.WriteAsync(list.ToJsonArray());
+        }
+
+        public string Name => "RequestHistory";
```

Comment:
Because the name of the resource has changed here, the demo client will be broken until we update it. If you would prefer to use the old name for now and create a task to bring the two inline, at least it will still work for now.